pipelines/test/vercheck.yaml

```diff
@@ -0,0 +1,137 @@
+    required: false
+    default: "auto"
+  match-type:
+    description: |
+      How to match the version string:
+      - 'contains': Check if version string appears anywhere in output (default)
+      - 'exact': Require exact match of entire output
+      - 'regex': Use version as regex pattern
+    required: false
+    default: "contains"
+
+# USAGE EXAMPLES:
+#
+# Minimal usage (uses package.version by default):
+#   - uses: vercheck
+#     with:
+#       bin: ${{vars.bin-name}}
+#
+# With custom version format:
+#   - uses: vercheck
+#     with:
+#       bin: nginx
+#       version: "nginx/${{package.version}}"
+#       version-flag: "-v"
+#
+# EXPECTED VARS:
+#   - bin-name: Primary binary name
+#
+pipeline:
+  - runs: |
+      vercheck() {
+        local binary="$1"
+        local expected_version="$2"
+        local version_flag="$3"
+        local match_type="$4"
+        local output exit_code
+
+        echo "   Checking version for binary: $binary"
+        echo "   Expected version: $expected_version"
+        echo "   Match type: $match_type"
+
+        # Check if binary exists in PATH
+        if ! command -v "$binary" >/dev/null 2>&1; then
+          echo "ERROR: Binary '$binary' not found in PATH"
+          echo "   Make sure the package installed correctly and binary is in PATH"
+          exit 1
+        fi
+        echo "Binary '$binary' found in PATH"
+
+        # Auto-detect version flag if not specified
+        if [ "$version_flag" = "auto" ]; then
+          echo "   Auto-detecting version flag..."
+          for flag in --version -version -V -v version; do
+            echo "   Trying: $binary $flag"
+            if output=$($binary $flag 2>&1) && [ $? -eq 0 ] && [ -n "$output" ]; then
+              version_flag="$flag"
+              echo "   Success with: $flag"
+              break
+            fi
+          done
+
+          if [ "$version_flag" = "auto" ]; then
+            echo "ERROR: Could not auto-detect version flag"
+            echo "   Tried: --version, -version, -V, -v, version"
+            exit 1
+          fi
+        else
+          echo "   Using specified version flag: $version_flag"
+          output=$($binary $version_flag 2>&1)
+          exit_code=$?
+
+          if [ $exit_code -ne 0 ]; then
+            echo "ERROR: '$binary $version_flag' failed with exit code $exit_code"
+            echo "   Command output:"
+            echo "$output" | sed 's/^/     /'
+            exit 1
+          fi
+        fi
+
+        # Perform version matching based on match-type
+        local match_success=false
+        case "$match_type" in
+          "exact")
+            if [ "$output" = "$expected_version" ]; then
+              match_success=true
+            fi
+            ;;
+          "regex")
+            if echo "$output" | grep -E "$expected_version" >/dev/null; then
+              match_success=true
+            fi
+            ;;
+          "contains"|*)
```

Comment:
don't allow bogus input.
it just means that when someone puts 'match_type' of 'world-cup', you have to support that forever.